### PR TITLE
Enhance AMIP and NCEP paperplots

### LINKED ISSUE #400: Enhance AMIP and NCEP paperplots
## Purpose 

We need to facilitate quantitative inspection of the NCEP and AMIP data. 

## To-do

- [x] return plot data
- [x] add diagnostics of total precipitable water (`q_liq_ice`) and turbulent energy surface fluxes (`tubulent_energy_fluxes`)

## QA
- [x] AMIP and NCEP plots in desired format

## Note
- alloc test limit was increased due to the additional accumulated diagnostics `allocs_limit["perf_coarse_single_modular"] = 18280800`

### PR DESCRIPTION
**Pull request by LenkaNovak**
Closes #400

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: experiments/AMIP/modular/coupler_driver_modular.jl [hunk 1]
```diff
@@ -610,20 +610,24 @@ if ClimaComms.iamroot(comms_ctx)
...
         )
 
         plot_spec = (;
             T = (; clims = (190, 320), units = "K"),
             u = (; clims = (-50, 50), units = "m/s"),
-            q_tot = (; clims = (0, 50), units = "g/kg"),
-            toa = (; clims = (-250, 210), units = "W/m^2"),
-            precipitation = (clims = (0, 1e-6), units = "kg/m^2/s"),
+            q_tot = (; clims = (0, 30), units = "g/kg"),
+            toa_fluxes = (; clims = (-250, 210), units = "W/m^2"),
```
Comment (akshaysridhar) [inline experiments/AMIP/modular/coupler_driver_modular.jl:624]: Would symmetric `clims` spec help with interpretation here ?

File: experiments/AMIP/modular/coupler_driver_modular.jl [hunk 2]
```diff
@@ -610,20 +610,24 @@ if ClimaComms.iamroot(comms_ctx)
...
+            precipitation = (clims = (0, 1e-4), units = "kg/m^2/s"),
             T_sfc = (clims = (225, 310), units = "K"),
+            q_liq_ice = (; clims = (0, 10), units = "g/kg"),
+            tubulent_energy_fluxes = (; clims = (-250, 210), units = "W/m^2"),
```
Comment (akshaysridhar) [inline experiments/AMIP/modular/coupler_driver_modular.jl:628]: Same as above comment on `clims` - would (-250,250) be useful here for visualizations?

File: experiments/AMIP/modular/coupler_driver_modular.jl [hunk 3]
```diff
@@ -353,7 +353,7 @@ dates = (; date = [date], date0 = [date0], date1 = [Dates.firstdayofmonth(date0)
 User can write custom diagnostics in the `user_diagnostics.jl`.
 =#
 monthly_3d_diags = init_diagnostics(
-    (:T, :u, :q_tot),
+    (:T, :u, :q_tot, :q_liq_ice),
```
Comment (akshaysridhar) [inline experiments/AMIP/modular/coupler_driver_modular.jl:356]: Variable changes : `q_liq_ice` (addition)  ,`toa_fluxes` (renaming), `turbulent_energy_fluxes` (addition) . each then requires an `operations` spec, which exists, plus use in a call to the `paperplots` function, which is also OK. Overall structure LGTM.

File: experiments/AMIP/modular/user_io/ncep_visualizer.jl [hunk 4]
```diff
@@ -50,21 +51,25 @@ function ncep_paperplots(
...
+
+        # create a named tuple with data
+        data = post_data.data
+        all_data = merge(all_data, [vname => data])
     end
 
     # combine plots and save figure
     save_fig = Plots.plot(
         all_plots...,
         size = (1500, 800),
```
Comment (akshaysridhar) [inline experiments/AMIP/modular/user_io/ncep_visualizer.jl:63]:
```suggestion
        size = (1500, 1200),
```

File: perf/flame.jl [hunk 5]
```diff
@@ -18,7 +18,7 @@ run_name_list =
 run_name = run_name_list[parse(Int, ARGS[2])]
 allocs_limit = Dict()
 allocs_limit["perf_default_modular_unthreaded"] = 8638304
-allocs_limit["perf_coarse_single_modular"] = 12671552
+allocs_limit["perf_coarse_single_modular"] = 18280800
```
Comment (akshaysridhar) [inline perf/flame.jl:21]: Seems to be a large increase - ~45%? Perhaps @nefrathenrici has an opinion on this?

File: experiments/AMIP/modular/coupler_driver_modular.jl (hunk 1 above)
Comment (LenkaNovak) [reply experiments/AMIP/modular/coupler_driver_modular.jl:624]: Sure, we can do that, as long as NCEP and AMIP are consistent and clear enough!

**Review by akshaysridhar — APPROVED** [review]

File: perf/flame.jl (hunk 5 above)
Comment (nefrathenrici) [reply perf/flame.jl:21]: The allocations increase shouldn't be coming from Atmos, I think @charleskawczynski has been able to decrease allocations in recent months. 
I assume monitoring heap allocations within the coupler must be more complicated than in atmos, so I don't know if this is an immediate issue.

File: experiments/AMIP/modular/user_io/user_diagnostics.jl [hunk 6]
```diff
@@ -68,7 +81,7 @@ end
 """
     get_var(cs::CoupledSimulation, ::Val{:precipitation})
```
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply experiments/AMIP/modular/user_io/user_diagnostics.jl:82]: Good point. Changed to `precipitation_rate`.

**Pull request merged**